packages/labs/analyzer/src/lib/lit-element/lit-element.ts

```diff
@@ -33,18 +39,75 @@ export const getLitElementDeclaration = (
   declaration: LitClassDeclaration,
   analyzer: AnalyzerInterface
 ): LitElementDeclaration => {
+  const reactiveProperties = getProperties(declaration, analyzer);
   return new LitElementDeclaration({
     tagname: getTagName(analyzer.typescript, declaration),
     // TODO(kschaaf): support anonymous class expressions when assigned to a const
     name: declaration.name?.text ?? '',
     node: declaration,
-    reactiveProperties: getProperties(declaration, analyzer),
+    reactiveProperties,
     ...getJSDocData(declaration, analyzer),
     getHeritage: () => getHeritage(declaration, analyzer),
-    ...getClassMembers(declaration, analyzer),
+    ...getLitElementClassMembers(declaration, analyzer, reactiveProperties),
   });
 };
 
+function getCustomElementField(
+  field: ClassField,
+  prop: ReactiveProperty
+): CustomElementField {
+  const {name} = field;
+  const {
+    privacy,
+    inheritedFrom,
+    source,
+    readonly,
+    type,
+    description,
+    deprecated,
+    summary,
+  } = field;
+  const attribute =
+    prop?.attribute === false
+      ? undefined
+      : typeof prop?.attribute === 'string'
+      ? prop.attribute
+      : name.toLowerCase();
+  const reflects = prop?.reflect ?? undefined;
+  return new CustomElementField({
+    name,
+    attribute,
+    reflects,
+    static: field.static,
+    privacy,
+    summary,
+    description,
+    deprecated,
+    default: field.default,
+    readonly,
+    inheritedFrom,
+    source,
+    type,
+  });
+}
+
+const getLitElementClassMembers = (
```

Comment:
Are the reactive properties not already fields when using decorators? How does this handle that case? Add that in a comment?

Reply:
It will narrow the type to CustomElementField, as we'd expect

What about when there's a static reactive prop without a corresponding class field. even if we'd recommend linting that out, nonetheless, the property will act like a field at runtime, right? so we should add it into the fields list

Reply:
added this logic and a test for it